packages/belos/src/BelosCGSingleRedIter.hpp

```diff
@@ -429,6 +429,9 @@ class CGSingleRedIter : virtual public CGIteration<ScalarType,MV,OP> {
     MVT::MvTransMv( one, *S_, *Z_, sHz );
     rHz = sHz(0,0);
     delta = sHz(1,0);
+    if ((delta < Teuchos::ScalarTraits<ScalarType>::eps()) &&
```

Comment:
Should the first test be less than or equal to, in case someone makes the (perhaps ill-advised) choice of using this solver for integer Scalar types?

Reply:
It looks like eps() isn't even defined for integer Scalar types, so this is probably going to bug out at compile time.

Reply:
A+